### PR TITLE
Fix shipping records w/o shopId

### PR DESCRIPTION
**Pull request by zenweasel**
Add shopId to all shipping records and removes many (but far from all) of the `shipping[0]` references.

This looks like a larger PR than it is because I migrated the `copyCartToOrder` method to its own file.

**Note** It should be super awesome if we could have Simple-Schema enforce that all shipping records need to contain a shopId but I could not get this to work. (I removed the `blackbox` record to make it validate, but it threw all sort of errors). I don't think this should block us now but I would like to revisit this as soon as we make the change to `node-simple-schema`.

### To Test

1. Create multiple shops
1. Add an item from each shop
1. Observe that there is a shipping records created for each shop that contains the shopId

### CODE REVIEW

File: server/methods/core/cartToOrder.js [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+  delete order.getTaxTotal;
+  delete order.getDiscounts;
+  delete order.getTotal;
+  delete order._id;
+
+
+  // Create a shipping record for each shop on the order
+  if (Array.isArray(order.shipping)) {
+    if (order.shipping.length > 0) {
+      const shippingRecords = [];
```
Comment (zenweasel) [inline server/methods/core/cartToOrder.js]: This is the code that was added here basically.

**Comment by spencern** [comment]
@zenweasel - I see you added a [WIP] tag recently. Should I wait to review?

**Comment by zenweasel** [comment]
@spencern Yeah, I spotted another reference to `shipping[0]` I need to fix. Will ping you when it's ready again.

File: server/methods/core/cartToOrder.js [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    item.workflow = {
+      status: "new",
+      workflow: ["coreOrderWorkflow/created"]
+    };
+
+    return item;
+  });
+
+  // Assign items to each shipping record based on the shopId of the item
+  _.each(order.items, (item) => {
```
Comment (zenweasel) [inline server/methods/core/cartToOrder.js]: And here, copying items to the shipping records, based upon their shopId

**Comment by zenweasel** [comment]
@spencern Should be ready now

File: server/methods/core/cartToOrder.js [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+  delete order.getTotal;
+  delete order._id;
+
+
+  // Create a shipping record for each shop on the order
+  if (Array.isArray(order.shipping)) {
+    if (order.shipping.length > 0) {
+      const shippingRecords = [];
+      order.shipping.map((shippingRecord) => {
+        const billingRecord = order.billing.find(billing => billing.shopId === shippingRecord.shopId);
```
Comment (spencern) [inline server/methods/core/cartToOrder.js]: This find only works with stripe unfortunately as other payment methods do not have a shopId.
Error below
```
Exception while invoking method 'cart/submitPayment' TypeError: Cannot read property '_id' of undefined
    at server/methods/core/cartToOrder.js:90:36
    at Array.map (native)
    at [object Object]._copyCartToOrder (server/methods/core/cartToOrder.js:88:22)
    at [object Object]._MethodHooks._wrappers.(anonymous function) (server/api/method-hooks.js:114:70)
    at packages/check.js:129:16
    at [object Object]._.extend.withValue (packages/meteor.js:1126:17)
    at Object.exports.Match._failIfArgumentsAreNotAllChecked (packages/check.js:128:41)
    at maybeAuditArgumentChecks (packages/ddp-server/livedata_server.js:1765:18)
    at packages/ddp-server/livedata_server.js:1686:15
    at [object Object]._.extend.withValue (packages/meteor.js:1126:17)
    at packages/ddp-server/livedata_server.js:1684:36
    at [object Object]._.extend.applyAsync (packages/ddp-server/livedata_server.js:1683:12)
    at [object Object]._.extend.apply (packages/ddp-server/livedata_server.js:1622:26)
    at [object Object]._.extend.call (packages/ddp-server/livedata_server.js:1604:17)
    at [object Object].<anonymous> (server/methods/core/hooks/cart.js:31:32)
    at server/api/method-hooks.js:122:36
 => awaited here:
    at Promise.await (/Users/sn/.meteor/packages/promise/.0.8.9.1o4y3g5++os+web.browser+web.cordova/npm/node_modules/meteor-promise/promise_server.js:60:12)
    at [object Object]._.extend.apply (packages/ddp-server/livedata_server.js:1635:22)
    at [object Object]._.extend.call (packages/ddp-server/livedata_server.js:1604:17)
    at [object Object].<anonymous> (server/methods/core/hooks/cart.js:31:32)
    at server/api/method-hooks.js:122:36
    at arrayEach (/Users/sn/dev/reaction-dev/node_modules/lodash/_arrayEach.js:15:9)
    at forEach (/Users/sn/dev/reaction-dev/node_modules/lodash/forEach.js:38:10)
    at [object Object]._MethodHooks._wrappers.(anonymous function) (server/api/method-hooks.js:121:5)
    at packages/check.js:129:16
    at [object Object]._.extend.withValue (packages/meteor.js:1126:17)
    at Object.exports.Match._failIfArgumentsAreNotAllChecked (packages/check.js:128:41)
    at maybeAuditArgumentChecks (packages/ddp-server/livedata_server.js:1765:18)
    at packages/ddp-server/livedata_server.js:719:19
    at [object Object]._.extend.withValue (packages/meteor.js:1126:17)
    at packages/ddp-server/livedata_server.js:717:46
    at [object Object]._.extend.withValue (packages/meteor.js:1126:17)
```

**Comment by zenweasel** [comment]
I'm not seeing any errors when processing orders with Authorize.net in a single-shop scenario.

**Comment by spencern** [comment]
Hmm, I'll try again

**Comment by spencern** [comment]
So the error I got was when I had two shops enabled and only checked out with the primary shop's item. Not really a use case for non-marketplace payment providers.

When I reset and checked out with a single shop, no errors were shown and the order was successful.

**Comment by spencern** [comment]
We probably _should_ add some more error handling around marketplace, but I don't think that's in scope for this ticket, so I'm going to go ahead and merge so we can keep moving.

**Pull request merged**